acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -737,9 +759,30 @@ fn any_witness_from_expression<F>(expr: &Expression<F>) -> Option<Witness> {
 pub(crate) fn is_predicate_false<F: AcirField>(
     witness: &WitnessMap<F>,
     predicate: &Option<Expression<F>>,
+    pedantic_solving: bool,
+    opcode_location: &ErrorLocation,
 ) -> Result<bool, OpcodeResolutionError<F>> {
     match predicate {
-        Some(pred) => get_value(pred, witness).map(|pred_value| pred_value.is_zero()),
+        Some(pred) => {
+            let pred_value = get_value(pred, witness);
+            if pedantic_solving {
+                pred_value.and_then(|predicate_value| {
+                    if predicate_value.is_zero() {
+                        Ok(true)
+                    } else if predicate_value.is_one() {
+                        Ok(false)
+                    } else {
+                        let opcode_location = *opcode_location;
+                        Err(OpcodeResolutionError::PredicateLargerThanOne {
```

Comment:
Is this an `Err` just because `is_predicate_false` already returned a `Result`, while other checks added `panic!` because we don't want to turn them into `Result`? Should this panic too, or is a predicate > 1 a user error?

Reply:
This is an error because I'm not sure whether it can occur and want to include the location so users can debug. It seems plausible that this could occur in low-level code or in the future if we add more low-level optimization helpers like `as_witness`.